lib/bracket-matcher.coffee

```diff
@@ -218,3 +224,6 @@ class BracketMatcher
 
   unsubscribe: ->
     @subscriptions.dispose()
+
+  getScopedSetting: (key) ->
+    atom.config.get(key, scope: @editor.getLastCursor().getScopeDescriptor())
```

Comment:
I think if a user wants to use scoped settings then he should be able to use them as much as he likes. For example, I want to disable bracket-matching in lisp files, but I still want to have it in comments and strings.

The [Flight Manual](http://flight-manual.atom.io/behind-atom/sections/scoped-settings-scopes-and-scope-descriptors/) doesn't really mention that most config options don't support scoped settings beyond language-specific ones (or at all), so a user (like myself) could easily think that features like bracket-matching fully support scopes.

On the other hand, even [editor-options](https://github.com/atom/atom/blob/master/src/text-editor.coffee#L3138) only support language-specific settings, so a user (like myself) could easily think that features like indentation are fully "scopeable" as well.

How ["hackable to the core—a tool you can customize to do anything"](https://atom.io/) do we want to be ?

The only downside I could think of is performance, but I noticed no problems/delays from that one additional config lookup.